gulp-tasks/utils/build-browser-bundle.js

```diff
@@ -36,21 +36,53 @@ const globals = (moduleId) => {
   }
 
   const packageName = splitModuleId.shift();
-  if (splitModuleId.length > 0) {
-    throw new Error(oneLine`
-    All imports of workbox-* modules must be done from the top level export.
-    (i.e. import * from 'workbox-*') This ensures that the browser
-    namespacing works correctly. Please remove '${splitModuleId.join('/')}'
-    from the import '${moduleId}'.
-  `);
+  const packagePath = path.join(__dirname, '..', '..', 'packages', packageName);
+  const namespacePath = splitModuleId.map((modulePiece) => {
+    // The browser namespace will need the file extension removed
+    return path.basename(modulePiece, path.extname(modulePiece));
+  }).join('.');
+  switch (splitModuleId.length) {
+    case 0: {
+      // Tried to pull in default export of module - this isn't allowed.
+      // A specific file must be referenced
+      throw new Error(oneLine`
+        You cannot use a module directly - you must specify a file, this is to
+        encourage best practice for tree shaking (i.e. only pulling in what you
+        use). Please fix '${moduleId}'
+      `);
+    }
+    case 1: {
+      // Pulling in a specific file is allowed as long as the exprts have
```

Comment:
Are you thinking there's a TODO here, or is the check somewhere else?

Reply:
I couldn't think of a way to import and test in the browser bundle (i.e. we don't have @std/esm loaded for this to work. At the moment there are tests to check for _private and I would duplicate that for public classes. This sadly doesn't capture examples where an import for an external module is requested, but that module doesn't actually exist - we could check for the actual file, but other than that, I'm kinda stumped.